taichi/ir/ir.cpp

```diff
@@ -18,6 +18,41 @@ IRBuilder &current_ast_builder() {
   return context->builder();
 }
 
+bool maybe_same_address(Stmt *var1, Stmt *var2) {
+  // Check if it's possible that two statements are pointing to the same address
+  if (var1 == var2)
+    return true;
+  if (var1->is<AllocaStmt>() || var2->is<AllocaStmt>())
```

Comment:
I think it's convenient to include other cases, so as to dig out possible optimization opportunities about global variables other than GlobalTemp and GlobalPtrs.

I'm very cautious about "other cases" in `OtherVariableOptimize`: whenever there is a global store, I presume it may store each of the variables.

Reply:
Now I understand. Thanks. Let's briefly document: return true when two statements might be the same address; false when two statements cannot be the same address.

Reply:
OK sounds good.